src/aspire/source/relion.py

```diff
@@ -87,25 +141,44 @@ def __init__(self, filepath, pixel_size=1, B=0, n_workers=-1, ignore_missing_fil
             If ignore_missing_files is True, then the first max_rows *available* rows from the .star file are
             considered.
         """
-
-        StarfileStack.__init__(self, filepath, n_workers=n_workers, ignore_missing_files=ignore_missing_files,
-                               max_rows=max_rows)
+        logger.debug(f'Creating ImageSource from starfile at path {filepath}')
 
         self.pixel_size = pixel_size
         self.B = B
+        self.n_workers = n_workers
 
-        rots = angles_to_rots(
-            self.df[['_rlnAngleRot_radians', '_rlnAngleTilt_radians', '_rlnAnglePsi_radians']].values
-        )
+        metadata = self.__class__.starfile2df(filepath, block_index_or_name, loop_index, ignore_missing_files, max_rows)
+
+        n = len(metadata)
+        if n == 0:
+            raise RuntimeError('No mrcs files found for starfile!')
+
+        # Peek into the first image and populate some attributes
+        first_mrc_filepath = metadata.iloc[0]['__mrc_filepath']
```

Comment:
Maybe the following lines (156–169) should be in a separate function?

Reply:
Opened a new issue #47 on this. Closing here.